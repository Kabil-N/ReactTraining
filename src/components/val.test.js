import {diff} from 'jest-diff';
import {render,screen} from '@testing-library/react'

// const {diff} = require('jest-diff');

test('sample test case',()=>{const a = {a: {b: {c: 5}}};
const b = {a: {b: {c: 6}}};

const result = diff(a, b);

// print diff
console.log(result);
})

const mockFunc = ()=>3
  
//   .mockReturnValue('default')
//   .mockImplementation(scalar => 42 + scalar)
//   .mockName('add42');

  test('rendered successfiully',()=>{
    // All calls and the name of the mock is written as a snapshot
    expect(mockFunc).toMatchSnapshot();
  })

  test('container',()=>{
    const table = document.createElement('h1')

    const { container,getByRole } = render(<div><h1>Hello</h1></div>, {
    container: document.body.appendChild(table),
    })

    // expect(container.firstChild.firstChild).toMatchInlineSnapshot('<h1>Hello</h1>');

    expect(getByRole('heading',{hidden:true})).toBeInTheDocument();

  })