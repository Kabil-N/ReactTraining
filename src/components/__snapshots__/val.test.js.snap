// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered successfiully 1`] = `[Function]`;
